### DISCUSSION TITLE
remove element from this.$el.remove() and how to decrease memory from browser also

### DISCUSSION THREAD

**Discussion by ericxin1982** [alpinejs/alpine#3916] [4. General]
Hi

I try to observe the memory change from this.$el.remove, but not sure whether it can achieve.

Thanks

**Comment by ekwoka** [comment]
It's hard to exactly do that.

What you can do to see if/when the element is garbage collected, you can do

```js
const logGC = new FinalizationRegistry((name) => console.log('Cleaned up element', name)


logGC.register(this.$el, 'My Component')
```

Of note: it not logging it does not necessarily mean it wasn't cleaned up, but if you do this a bunch and put memory pressure on the environment and it never logs no matter how many times you try, it definitely isn't getting cleaned up.

**Comment by ericxin1982** [comment]
Thanks
